### ISSUE TITLE
HTTP/2 bi-directional connectivity for real-time apps

### ISSUE THREAD

**Issue by g105b** [PhpGt/WebEngine#304]
WebSockets were a great tool for implementing real-time updates without the long-polling of XHR, but now HTTP/2 is a thing, we can achieve the same level of instant bi-directional communication between server and client with the HTTP protocol itself.

HTTP/2 allows for TCP connections to be kept open between server and client, allowing for the server to push new updates to the page without requiring any extra libraries.

To get the benefits of real time applications without breaking the principal of single logic authority, WebEngine should have the ability to render the changed parts of the page over HTTP/2 persistent connections, requiring very minimal JavaScript usage on the client.

**Comment by g105b** [comment]
This idea in this issue is to be able to have a WebEngine way of pushing updates to the view after the first render has completed. This will be possible after #307, but there is a lot of change required to the Lifecycle object to achieve this, so definitely won't be in v3.

**Comment by g105b** [comment]
Out of scope.
